crowbar_framework/app/models/dns_service.rb

```diff
@@ -49,6 +49,22 @@ def validate_proposal_after_save proposal
       validation_error("At least one nameserver or one node with the dns-server role must be specified.")
     end
 
+    proposal["attributes"]["dns"]["records"].each do |host, records|
+      unless ["A", "CNAME"].include?(records[:type])
+        validation_error I18n.t(
+          "barclamp.#{@bc_name}.validation.invalid_record_type",
+          type: records[:type],
+          name: host
+        )
+      end
+      if records[:type] == "CNAME" && records[:values].length > 1
```

Comment:
I beg to disagree: this makes the code less readable in this case since we have a bunch of checks, and each check can generate an error.